sw/device/lib/runtime/pmp.c

```diff
@@ -0,0 +1,539 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/runtime/pmp.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/stdasm.h"
+
+// "Volume II: RISC-V Privileged Architectures V20190608-Priv-MSU-Ratified",
+// "3.6.1 Physical Memory Protection CSRs",
+// "Figure 3.28: PMP configuration register format".
+#define PMP_CFG_CSR_R 0
```

Comment:
If these don't need to be defines, make them constants (with clearer names, like `Read` rather than `R`.)

Reply:
These correspond directly to PMP specification, so I would prefer to keep them as they are (regarding the s/R/READ/).

I don't see any value in making these defines into constants.

Reply:
> I don't see any value in making these defines into constants.

Macros-as-constants are not a pattern we should be using when we can avoid it. Macros pollute the token namespace and are untyped, leading to awkwardness around integer promotion.

Consider cleaning these up in a followup.

Also, there's no reason to stick to the spec's names when we can come up with something more readable, and write something like the following:
```c
/**
 * Corresponds to the `PMP_CFG_CSR_R` PMP constant.
 */
static const bitfield_bit32_index_t kPmpReadBit = 0;
```